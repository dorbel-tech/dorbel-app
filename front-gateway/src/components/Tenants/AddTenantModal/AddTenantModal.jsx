import React from 'react';
import autobind from 'react-autobind';
import { inject } from 'mobx-react';
import { Button } from 'react-bootstrap';
import FormWrapper, { FRC } from '~/components/FormWrapper/FormWrapper';

@inject('appProviders')
export default class AddTenantModal extends React.Component {
  static title = 'הוספת דייר חדש לדירה';

  constructor(props) {
    super(props);
    autobind(this);
  }

  submit() {
    const { formsy } = this.refs.form.refs;

    if (formsy.state.isValid) {
      const { appProviders, listing_id } = this.props;
      appProviders.listingsProvider.addTenant(listing_id, formsy.getCurrentValues())
        .catch(() => appProviders.notificationProvider.error('חלה שגיאה בהוספת הדייר'))
        .then(() => appProviders.modalProvider.close());
    } else {
      formsy.submit(); // show validation errors
    }
  }

  render() {
    return (
      <div>
        <p>
          <b>צרפו דייר חדש לדירה שהשכרתם</b><br/>
          הדייר יקבל מייל עם הזמנה למלא את פרטיו, שיתווספו עבורכם אוטומטית למסך ניהול הדירה
        </p>
        <FormWrapper.Wrapper layout="vertical" ref="form">
          <FRC.Input value="" name="email" type="email" label="אי-מייל" validations="isEmail" />
          <FRC.Input value="" name="first_name" type="text" label="שם פרטי" required />
          <FRC.Input value="" name="last_name" type="text" label="שם משפחה" />
          <FRC.Input value="" name="phone" type="text" label="טלפון" />
          <Button bsStyle="success" onClick={this.submit} block>אישור</Button>
        </FormWrapper.Wrapper>
      </div>
    );
  }
}

AddTenantModal.wrappedComponent.propTypes = {
  appProviders: React.PropTypes.object.isRequired,
  listing_id: React.PropTypes.number.isRequired
};
