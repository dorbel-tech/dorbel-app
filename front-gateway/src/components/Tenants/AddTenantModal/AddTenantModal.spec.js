import React from 'react';
import { mount } from 'enzyme';
import faker from 'faker';

import AddTenantModal from '~/components/Tenants/AddTenantModal/AddTenantModal';
import { Button } from 'react-bootstrap';

describe('Add Tenant Modal', () => {
  let appProvidersMock;
  beforeEach(() => {
    appProvidersMock = {
      listingsProvider: {
        addTenant: jest.fn()
      },
      modalProvider: {
        close: jest.fn()
      }
    };
  });

  const addTenantModal = (listing_id) => mount(<AddTenantModal.wrappedComponent listing_id={listing_id} appProviders={appProvidersMock} />);

  it('should match snapshot in empty state', () => {
    const listing_id = 58134; // needs to be static for the snapshot
    const wrapper = addTenantModal(listing_id);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not call provider if form is empty', () => {
    const wrapper = mount(<AddTenantModal.wrappedComponent listing_id={faker.random.number()} appProviders={appProvidersMock} />);
    wrapper.find(Button).simulate('click');
    expect(appProvidersMock.listingsProvider.addTenant).not.toHaveBeenCalled();
  });

  it('send form values to provider', () => {
    const listing_id = faker.random.number();
    const wrapper = mount(<AddTenantModal.wrappedComponent listing_id={listing_id} appProviders={appProvidersMock} />);
    const first_name = faker.name.firstName();
    const firstNameInput = wrapper.node.form.formsy.inputs.find(input => input.props.name === 'first_name');
    firstNameInput.setValue(first_name);
    appProvidersMock.listingsProvider.addTenant.mockReturnValue(Promise.resolve());
    wrapper.find(Button).simulate('click');

    const addTenantCallArgs = appProvidersMock.listingsProvider.addTenant.mock.calls[0];
    expect(addTenantCallArgs[0]).toBe(listing_id);
    expect(addTenantCallArgs[1]).toHaveProperty('first_name', first_name);
  });
});
