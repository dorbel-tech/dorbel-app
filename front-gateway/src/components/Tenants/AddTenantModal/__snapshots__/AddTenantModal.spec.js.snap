exports[`Add Tenant Modal should match snapshot in empty state 1`] = `
<AddTenantModal
  appProviders={
    Object {
      "listingsProvider": Object {
        "addTenant": [Function],
      },
      "modalProvider": Object {
        "close": [Function],
      },
    }
  }
  listing_id={58134}>
  <div>
    <p>
      <b>
        צרפו דייר חדש לדירה שהשכרתם
      </b>
      <br />
      הדייר יקבל מייל עם הזמנה למלא את פרטיו, שיתווספו עבורכם אוטומטית למסך ניהול הדירה
    </p>
    <FormWrapper
      layout="vertical">
      <Formsy
        className="form-vertical"
        onChange={[Function]}
        onError={[Function]}
        onInvalid={[Function]}
        onInvalidSubmit={[Function]}
        onSubmit={[Function]}
        onSuccess={[Function]}
        onValid={[Function]}
        onValidSubmit={[Function]}
        preventExternalInvalidation={false}
        validationErrors={null}>
        <form
          className="form-vertical"
          onSubmit={[Function]}>
          <Input
            addonAfter={null}
            addonBefore={null}
            buttonAfter={null}
            buttonBefore={null}
            disabled={false}
            label="אי-מייל"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="email"
            validateOnSubmit={false}
            validatePristine={false}
            validationError=""
            validationErrors={Object {}}
            validations="isEmail"
            value="">
            <Row
              elementWrapperClassName={
                Array [
                  "",
                  undefined,
                ]
              }
              fakeLabel={false}
              hasErrors={false}
              htmlFor="frc-email-1987260706"
              label="אי-מייל"
              labelClassName={
                Array [
                  "",
                  undefined,
                ]
              }
              layout="vertical"
              required={false}
              rowClassName={
                Array [
                  "",
                  undefined,
                ]
              }>
              <div
                className="form-group">
                <label
                  className="control-label"
                  data-required={false}
                  htmlFor="frc-email-1987260706">
                  אי-מייל
                </label>
                <input
                  className="form-control"
                  disabled={false}
                  id="frc-email-1987260706"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="email"
                  value="" />
              </div>
            </Row>
          </Input>
          <Input
            addonAfter={null}
            addonBefore={null}
            buttonAfter={null}
            buttonBefore={null}
            disabled={false}
            label="שם פרטי"
            name="first_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            validateOnSubmit={false}
            validatePristine={false}
            validationError=""
            validationErrors={Object {}}
            value="">
            <Row
              elementWrapperClassName={
                Array [
                  "",
                  undefined,
                ]
              }
              fakeLabel={false}
              hasErrors={false}
              htmlFor="frc-first_name-1496701523"
              label="שם פרטי"
              labelClassName={
                Array [
                  "",
                  undefined,
                ]
              }
              layout="vertical"
              required={true}
              rowClassName={
                Array [
                  "",
                  undefined,
                ]
              }>
              <div
                className="form-group">
                <label
                  className="control-label"
                  data-required={true}
                  htmlFor="frc-first_name-1496701523">
                  שם פרטי
                  <span
                    className="required-symbol">
                     *
                  </span>
                </label>
                <input
                  className="form-control"
                  disabled={false}
                  id="frc-first_name-1496701523"
                  name="first_name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={true}
                  type="text"
                  value="" />
              </div>
            </Row>
          </Input>
          <Input
            addonAfter={null}
            addonBefore={null}
            buttonAfter={null}
            buttonBefore={null}
            disabled={false}
            label="שם משפחה"
            name="last_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            validateOnSubmit={false}
            validatePristine={false}
            validationError=""
            validationErrors={Object {}}
            value="">
            <Row
              elementWrapperClassName={
                Array [
                  "",
                  undefined,
                ]
              }
              fakeLabel={false}
              hasErrors={false}
              htmlFor="frc-last_name--1017343626"
              label="שם משפחה"
              labelClassName={
                Array [
                  "",
                  undefined,
                ]
              }
              layout="vertical"
              required={false}
              rowClassName={
                Array [
                  "",
                  undefined,
                ]
              }>
              <div
                className="form-group">
                <label
                  className="control-label"
                  data-required={false}
                  htmlFor="frc-last_name--1017343626">
                  שם משפחה
                </label>
                <input
                  className="form-control"
                  disabled={false}
                  id="frc-last_name--1017343626"
                  name="last_name"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text"
                  value="" />
              </div>
            </Row>
          </Input>
          <Input
            addonAfter={null}
            addonBefore={null}
            buttonAfter={null}
            buttonBefore={null}
            disabled={false}
            label="טלפון"
            name="phone"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            type="text"
            validateOnSubmit={false}
            validatePristine={false}
            validationError=""
            validationErrors={Object {}}
            value="">
            <Row
              elementWrapperClassName={
                Array [
                  "",
                  undefined,
                ]
              }
              fakeLabel={false}
              hasErrors={false}
              htmlFor="frc-phone-1421578746"
              label="טלפון"
              labelClassName={
                Array [
                  "",
                  undefined,
                ]
              }
              layout="vertical"
              required={false}
              rowClassName={
                Array [
                  "",
                  undefined,
                ]
              }>
              <div
                className="form-group">
                <label
                  className="control-label"
                  data-required={false}
                  htmlFor="frc-phone-1421578746">
                  טלפון
                </label>
                <input
                  className="form-control"
                  disabled={false}
                  id="frc-phone-1421578746"
                  name="phone"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="text"
                  value="" />
              </div>
            </Row>
          </Input>
          <Button
            active={false}
            block={true}
            bsClass="btn"
            bsStyle="success"
            disabled={false}
            onClick={[Function]}>
            <button
              className="btn btn-success btn-block"
              disabled={false}
              onClick={[Function]}
              type="button">
              אישור
            </button>
          </Button>
        </form>
      </Formsy>
    </FormWrapper>
  </div>
</AddTenantModal>
`;
